pkg/generate/generate.go

```diff
@@ -472,6 +466,38 @@ func manageClone(log logr.Logger, apiVersion, kind, namespace, name string, clon
 		return nil, Skip, fmt.Errorf("source resource %s %s/%s/%s not found. %v", apiVersion, kind, rNamespace, rName, err)
 	}
 
+	updateSource := true
+
+	// add label
+	label := obj.GetLabels()
+	if len(label) == 0 {
+		label = make(map[string]string)
+		label["generate.kyverno.io/clone-policy-name"] = policy
+	} else {
+		if label["generate.kyverno.io/clone-policy-name"] != "" {
+			policyNames := label["generate.kyverno.io/clone-policy-name"]
+			if !strings.Contains(policyNames, policy) {
+				policyNames = policyNames + "," + policy
+				label["generate.kyverno.io/clone-policy-name"] = policyNames
+			} else {
+				updateSource = false
+			}
+		} else {
+			label["generate.kyverno.io/clone-policy-name"] = policy
+		}
+	}
+
+	if updateSource {
+		log.V(4).Info("updating existing clone source")
+		obj.SetLabels(label)
+		_, err = client.UpdateResource(apiVersion, kind, rNamespace, obj, false)
+		if err != nil {
+			logger.Error(err, "updating existing resource")
```

Comment:
change "updating existing resource" to "failed to update source resource" and add resource details